rp235x-hal/src/gpio/pin.rs

```diff
@@ -168,6 +168,7 @@ pub(crate) fn set_function<P: PinId>(pin: &P, function: DynFunction) {
         DynFunction::Pio1 => FUNCSEL_A::PIO1,
         DynFunction::Pio2 => FUNCSEL_A::PIO2,
         DynFunction::Clock => FUNCSEL_A::GPCK,
+        DynFunction::XipCs1 => FUNCSEL_A::GPCK,
```

Comment:
The rp235x-pac is missing function select definition for XipCS1:
https://github.com/rp-rs/rp235x-pac/blob/main/svd/RP2350.yaml#L321

But FUNCSEL_A::GPCK has the same value (9). We should add a variant specific for XipCS1.

Reply:
OK I'll go fix turf I'll add a ticket for that

Reply:
https://github.com/rp-rs/rp235x-pac/issues/7

It's not an easy fix, sadly.

Reply:
We should probably limit which pins you can set with ::Clock and ::XipCs1, to avoid surprises.

Reply:
Aren't pins that can be set as XipCs1 already being limited by:
https://github.com/rp-rs/rp-hal/pull/873/files#diff-514435c8bc1dac690be86445ca032b00cea05a617a1fef193197402cf99d94e0R174
and 
https://github.com/rp-rs/rp-hal/pull/873/files#diff-514435c8bc1dac690be86445ca032b00cea05a617a1fef193197402cf99d94e0R211
?

Reply:
Yup, I missed that. It seems fine.